fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java

```diff
@@ -192,12 +221,13 @@ public Plan plan(LogicalPlan plan, PhysicalProperties requireProperties, Explain
             }
 
             Optional<ScheduledExecutorService> timeoutExecutor = Optional.empty();
-            if (ConnectContext.get().getSessionVariable().enableNereidsTimeout) {
+            if (statementContext.getConnectContext().getSessionVariable().enableNereidsTimeout) {
```

Comment:
we can add a private method `getSessionVariable()`

```java
statementContext.getConnectContext().getSessionVariable()
```

Reply:
yes, how to get session variable and connect context can be refactor in another pr